docs/content/latest/api/ysql/exprs/func_yb_hash_code.md

```diff
@@ -0,0 +1,195 @@
+ Index Scan using sample_table_pkey on sample_table  (cost=0.00..5.88 rows=16 width=12) (actual time=2.867..2.919 rows=18 loops=1)
+   Index Cond: ((yb_hash_code(x, y) <= 128) AND (yb_hash_code(x, y) >= 0))
+ Planning Time: 0.052 ms
+ Execution Time: 2.953 ms
+(4 rows)
+```
+
+Here, we see that the primary key index was used and no row was rechecked at the postgres layer.
+
+As the `yb_hash_code` calls in this statement request the hash code of `x` and `y`, or in other words the full hash key of `sample_table`, we push down these calls. This pushdown is done by specifying a requested hash value range in the DocDB RPC request. As an added side effect, the RPC request will only be sent out to tablet servers that definitely contain values in the requested hash range. 
+
+This pushdown functionality works for secondary indexes too. This is a feature that is currently unavailable with the YCQL counterpart of this function, partition_hash(). For example, if we create an index on `sample_table` as follows:
+
+```
+CREATE INDEX sample_idx ON sample_table ((x, z) HASH);
+```
+
+We can consider a modified version of the above `SELECT` query as follows:
+
+```
+SELECT * FROM sample_table WHERE yb_hash_code(x,z) <= 128 AND yb_hash_code(x,z) >= 0;
+```
+
+Here, the `yb_hash_code` calls are on `x` and `z`. `x` and `z` do not form the full hash key of the base sample_table table but they do form the full hash key of the index table, `sample_idx`. Hence, the optimizer will consider pushing down the calls to the DocDB layer if an index scan using `sample_idx` is chosen. Note that the optimizer may choose not to go with a secondary scan if it deems the requested hash range to be large enough to warrant doing a simple full table scan instead. The `EXPLAIN ANALYZE` result for this could be as follows
+
+```
+                                                         QUERY PLAN                                                         
+----------------------------------------------------------------------------------------------------------------------------
+ Index Scan using sample_idx on sample_table  (cost=0.00..5.96 rows=16 width=12) (actual time=8.923..8.975 rows=18 loops=1)
+   Index Cond: (yb_hash_code(x, z) <= 128)
+ Planning Time: 0.066 ms
+ Execution Time: 9.033 ms
+(4 rows)
+```
+
+Note that we can also use [pg_hint_plan](../../../../explore/query-1-performance/pg-hint-plan/) to manipulate the index that is used.
+We create a duplicate index of `sample_idx`:
+
+```
+CREATE INDEX sample_idx_dup ON sample_table ((x,z) HASH);
+```
+
+And then we execute the same query with a pg_hint_plan directive:
+
+```
+SET pg_hint_plan.enable_hint=ON;
+/*+IndexScan(sample_table sample_idx_dup) */
+EXPLAIN ANALYZE SELECT * FROM sample_table WHERE yb_hash_code(x,z) <= 128 AND yb_hash_code(x,z) >= 0;
+                                                          QUERY PLAN                                                          
```

Comment:
Split the inputs and outputs
````suggestion
```sql
SET pg_hint_plan.enable_hint=ON;
/*+IndexScan(sample_table sample_idx_dup) */
EXPLAIN ANALYZE SELECT * FROM sample_table WHERE yb_hash_code(x,z) <= 128 AND yb_hash_code(x,z) >= 0;
```

```output
                                                          QUERY PLAN
````